pkg/sqlutils/transactions.go

```diff
@@ -7,12 +7,9 @@ import (
 	"github.com/jmoiron/sqlx"
 )
 
-// TransactionFunc defines the function signature needed to represent passing a transaction and returning a generic type
-type TransactionFunc[T any] func(*sqlx.Tx) (*T, error)
-
-// WithTransaction is a wrapper function which handles creating, committing or rolling back a transaction
+// WithTransactionRet is a wrapper function which handles creating, committing or rolling back a transaction
 // If there are any errors when executing the txFunc, the tx is rolled back. Otherwise, the tx is committed.
-func WithTransaction[T any](txPrep TransactionPreparer, txFunc TransactionFunc[T]) (*T, error) {
+func WithTransactionRet[T any](txPrep TransactionPreparer, txFunc func(*sqlx.Tx) (*T, error)) (*T, error) {
```

Comment:
Just out of curiosity, what is the purpose of this additional function vs WithTransaction as it stood?

Reply:
Great question! I found myself using `_, err := WithTransaction[any](... , func(tx) (*any, error))` in places where the transaction was purely executional (i.e., no return value required) or where I could not get `WithTransaction` to return the desired type (a slice of pointers, for example)

Reply:
Ah! Makes sense, thanks Sam! I'd be curious about the slice of pointers issue if you ever want to pair on that!

Reply:
Would love to! 😄